Replaced multiple if statements with regex switch statements

HCRitter opened the pull request:
Refactored code to improve architecture determination:
- Replaced multiple if statements with regex switch statements for better readability and maintainability.
- Initially, the architecture is determined based on the hints in the 'title' property. If any hints match, the architecture is assigned accordingly. Otherwise, it is set to $null.
- If the architecture is still null, a fallback mechanism has been implemented. The 'link' property is then used to further determine the architecture.
- These changes simplify the code and enhance its flexibility in identifying the architecture.

Hey Chrissy, I hope you are fine with my improvements.
To test them you can check this code:
```PowerShell


$titles = @(
    "Intel IA32 Processor",
    "Intel IA64 Processor",
    "Intel 64-Bit Processor",
    "Intel 32-Bit Processor",
    "AMD64 Processor",
    "Intel x64 Processor",
    "Intel x86 Processor",
    "ARM64 Processor",
    "ARM-based Processor"
)

foreach ($title in $titles) {
    $item = [PSCustomObject]@{
        title = $title
        Architecture = $null
    }

    $item.Architecture = switch -regex ($item.title) {
        ".*ia32.*" { "IA32" }
        ".*ia64.*" { "IA64" }
        ".*(32-Bit|x86).*" { "x86" }
        ".*(x64|AMD64|64-Bit).*" { "x64" }
        ".*ARM64.*" { "ARM64" }
        ".*ARM-based.*" { "ARM32" }
        default { $null } # Handle the case if no matching architecture is found
    }
    

    Write-Host "Title: $($item.title) | Architecture: $($item.Architecture)"
}
```

If you like I could check for other improvements as well :)

HCRitter committed:
Replaced multiple if statements with regex switch statements

Refactored code to improve architecture determination:
- Replaced multiple if statements with regex switch statements for better readability and maintainability.
- Initially, the architecture is determined based on the hints in the 'title' property. If any hints match, the architecture is assigned accordingly. Otherwise, it is set to $null.
- If the architecture is still null, a fallback mechanism has been implemented. The 'link' property is then used to further determine the architecture.
- These changes simplify the code and enhance its flexibility in identifying the architecture.

HCRitter commented:
I have seen there are multiple other instances of the if checks for the architecture. in other functions in that file like:
```PowerShell
if ($title -match "ia32") {
                        $arch = "IA32"
                    }
                    if ($title -match "ia64") {
                        $arch = "IA64"
                    }
                    if ($title -match "64-Bit" -and $title -notmatch "32-Bit" -and -not $arch) {
                        $arch = "x64"
                    }
                    if ($title -notmatch "64-Bit" -and $title -match "32-Bit" -and -not $arch) {
                        $arch = "x86"
                    }
                    if ($title -match "x64" -or $title -match "AMD64") {
                        $arch = "x64"
                    }
                    if ($title -match "x86") {
                        $arch = "x86"
                    }
                    if ($title -match "ARM64") {
                        $arch = "ARM64"
                    }
                    if ($title -match "ARM-based") {
                        $arch = "ARM32"
                    }
#...
```
If you like I can change them as well.

potatoqualitee commented:
awesome! thank you. i havent had time in forever to work on this project and need to set aside some maintenance. this is a great change but may take some time to get done

potatoqualitee commented:
yes, please do change. i prefer pretty and dont know regex 🚀

HCRitter committed:
Update Get-KbUpdate.ps1

Refactored architecture if statement handling in other occurrences

In this commit, I refactored the handling of the architecture if statement in other parts of the codebase. This change ensures consistent behavior and improves code maintainability.

HCRitter commented:
@potatoqualitee now it should be nice and clean :D

HCRitter commented:
@potatoqualitee anything else to change? Should be good to go now :)
